### PR TITLE
424 session expired bug

### PR DESCRIPTION
**Pull request by acaruso-oddball**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:
Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

This PR resolves a bug encountered when a user declines the Terms of Use on the VA.gov production environment, where they are incorrectly redirected to the "Session Expired" page instead of the intended "Terms of Use Declined" page. To fix this issue, updates were made to the apiRequest function to ensure proper handling of the logout flow and redirection after declining Terms of Use. New test cases were also implemented to validate the correct redirection behavior and prevent future regressions.

[Jira Ticket](https://jira.devops.va.gov/browse/VI-424)

## Testing done
Additional tests have been added to ensure proper functionality of apiRequest. 

## Screenshots
Update not able to be captured in this format. 

## What areas of the site does it impact?

Redirect after Terms of Use decline. 

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by CBonade — APPROVED** [review]

**Review by CBonade — APPROVED** [review]

File: src/platform/utilities/api/index.js [hunk 1]
```diff
@@ -79,20 +79,31 @@ export function fetchAndUpdateSessionExpiration(url, settings) {
...
+ * whether the code is running in production or non-production mode. If no environment object is provided, the function defaults to using the
+ * global `environment` object.
  */
-export function apiRequest(resource, optionalSettings, success, error) {
+export function apiRequest(
+  resource,
+  optionalSettings,
+  success,
+  error,
+  env = environment,
```
Comment (asg5704) [inline src/platform/utilities/api/index.js:100]: This parameter can be removed.

File: src/platform/utilities/api/index.js [hunk 2]
```diff
@@ -79,20 +79,31 @@ export function fetchAndUpdateSessionExpiration(url, settings) {
...
+) {
   const apiVersion = (optionalSettings && optionalSettings.apiVersion) || 'v0';
   const baseUrl = `${environment.API_URL}/${apiVersion}`;
   const url = resource[0] === '/' ? [baseUrl, resource].join('') : resource;
   const csrfTokenStored = localStorage.getItem('csrfToken');
+  const isProd = env.isProduction();
```
Comment (asg5704) [inline src/platform/utilities/api/index.js:106]: Let's just use `environment.isProduction()`

File: src/platform/utilities/tests/api/index.unit.spec.js [hunk 3]
```diff
@@ -28,9 +28,150 @@ describe('test wrapper', () => {
...
+
+      sessionStorage.setItem('shouldRedirectExpiredSession', 'true');
+
+      Object.defineProperty(window, 'location', {
+        value: {
+          pathname: '/terms-of-use/declined',
+          assign: sinon.stub(),
+        },
+        writable: true,
+      });
```
Comment (asg5704) [inline src/platform/utilities/tests/api/index.unit.spec.js:139]: I'm not a fan of this. It should be fine for now but we should really think about alternatives.

File: src/platform/utilities/api/index.js [hunk 4]
```diff
@@ -145,15 +156,17 @@ export function apiRequest(resource, optionalSettings, success, error) {
...
-      if (environment.isProduction()) {
+      if (isProd) {
         const { pathname } = window.location;
 
         const shouldRedirectToSessionExpired =
           response.status === 401 &&
           !pathname.includes('auth/login/callback') &&
-          sessionStorage.getItem('shouldRedirectExpiredSession') === 'true';
+          sessionStorage.getItem('shouldRedirectExpiredSession') === 'true' &&
+          !pathname.includes('/terms-of-use/declined');
```
Comment (asg5704) [inline src/platform/utilities/api/index.js:166]: Have you tested this manually to see if it works? I'm unable to get it to redirect properly when locally changing the `isProd = true`

**Review by asg5704 — APPROVED** [review]

**Pull request merged**